test/jdk/java/awt/event/KeyEvent/KeyChar/KeyCharTest.java

```diff
@@ -58,21 +62,38 @@ public void eventDispatched(AWTEvent event) {
     }
 
     public static void main(String[] args) throws Exception {
-
-
         Frame frame = new Frame();
         frame.setSize(300, 300);
+        frame.setLocationRelativeTo(null);
         frame.setVisible(true);
+        frame.toFront();
+        frame.requestFocus();
+
         Robot robot = new Robot();
-        robot.setAutoDelay(50);
+        robot.setAutoDelay(100);
         robot.waitForIdle();
-
+        robot.delay(1000);
 
         robot.keyPress(KeyEvent.VK_DELETE);
         robot.keyRelease(KeyEvent.VK_DELETE);
         robot.waitForIdle();
+        robot.delay(1000);
+
+        if (eventsCount != 3) {
+            // We are going to fail
+            Point loc = frame.getLocationOnScreen();
+            GraphicsDevice gds[] = GraphicsEnvironment.getLocalGraphicsEnvironment().getScreenDevices();
+            for (GraphicsDevice gd : gds) {
+                if (gd.getDefaultConfiguration().getBounds().contains(loc)) {
+                    BufferedImage capture = robot.createScreenCapture(gd.getDefaultConfiguration().getBounds());
+                    File captureFile = new File("capture.png");
+                    ImageIO.write(capture, "png", captureFile);
```

Comment:
Can you confirm this png gets written somewhere that isn't scratch spaced deleted at the end of testing ?
I don't think we do this multi-screen dance in most tests since the default screen should be good enough.
Why isn't it here ?

Reply:
I can confirm that in case of the failure the file is accessible in the work directory for download. I added this multi-monitor just in case if test fails because window is in the strange spot on multi-monitor configuration so i want to capture which screen it is on. That's not necessary since in 99.99% cases it will stay on the default display, but it does not hurt either.

Reply:
You can ask the GraphcsDevice where the Frame is located from the Frame itself, not necessary to check the bounds etc.